pkg/pillar/cmd/zedkube/kubeservice.go

```diff
@@ -0,0 +1,334 @@
+	// List all ingresses across all namespaces
+	ingresses, err := clientset.NetworkingV1().Ingresses("").List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("GetAllKubeIngresses: can't get ingresses: %v", err)
+		return nil, err
+	}
+
+	var ingressInfoList []types.KubeIngressInfo
+
+	for _, ing := range ingresses.Items {
+		// Skip ingresses in excluded namespaces
+		if _, excluded := excludedKubeNamespaces[ing.Namespace]; excluded {
+			continue
+		}
+
+		// Get the IP addresses of the ingress (if any)
+		var ingressIPs []string
+		for _, lbIngress := range ing.Status.LoadBalancer.Ingress {
+			if lbIngress.IP != "" {
+				// Check if the IP is on one of our interfaces and log accordingly
+				isDeviceIP := z.isDeviceInterfaceIP(lbIngress.IP)
+				if isDeviceIP {
+					log.Functionf("Ingress %s/%s includes IP %s that matches a device interface",
+						ing.Namespace, ing.Name, lbIngress.IP)
+					// Only add the IP if it belongs to one of our interfaces
+					ingressIPs = append(ingressIPs, lbIngress.IP)
+				}
+			} else if lbIngress.Hostname != "" {
+				ingressIPs = append(ingressIPs, lbIngress.Hostname) // Use hostname if IP is not available
+			}
+		}
+
+		// Sort the IPs to ensure consistent order
+		if len(ingressIPs) > 1 {
+			sort.Strings(ingressIPs)
+			log.Functionf("Sorted %d ingress IPs for %s/%s", len(ingressIPs), ing.Namespace, ing.Name)
+		}
+
+		// Process each rule in the ingress
+		for _, rule := range ing.Spec.Rules {
+			hostname := rule.Host
+			if rule.HTTP == nil {
+				continue
+			}
+
+			// Process each path in the rule
+			for _, path := range rule.HTTP.Paths {
+				if path.Backend.Service == nil {
+					continue
+				}
+
+				// Default to HTTP, check for TLS configuration
+				protocol := "http"
+				for _, tls := range ing.Spec.TLS {
+					for _, host := range tls.Hosts {
+						if host == hostname || host == "" {
+							protocol = "https"
+							break
+						}
+					}
+					if protocol == "https" {
+						break
+					}
+				}
+
+				// Create ingress info object
+				pathType := ""
+				if path.PathType != nil {
+					pathType = string(*path.PathType)
+				}
+
+				// Get the service type by looking up the service in already fetched serviceInfoList
+				var serviceType corev1.ServiceType = corev1.ServiceTypeClusterIP // default if not found
+				serviceName := path.Backend.Service.Name
+				serviceNamespace := ing.Namespace
+
+				// Look for the service in our already fetched serviceInfoList
+				serviceFound := false
+				for _, svcInfo := range serviceInfoList {
+					if svcInfo.Name == serviceName && svcInfo.Namespace == serviceNamespace {
+						serviceType = svcInfo.Type
+						serviceFound = true
+
+						// Add more detailed logging for LoadBalancer and NodePort services
+						if serviceType == corev1.ServiceTypeLoadBalancer {
+							log.Noticef("GetAllKubeIngresses: found LoadBalancer service %s/%s (LB IP: %s)",
```

Comment:
Is this logged for every service every 30 seconds?

Reply:
Forgot to change them into debug type. will change them.